crates/ide_assists/src/handlers/move_module_to_file.rs

```diff
@@ -65,7 +65,25 @@ pub(crate) fn move_module_to_file(acc: &mut Assists, ctx: &AssistContext) -> Opt
             }
             format_to!(buf, "mod {};", module_name);
 
-            builder.replace(module_ast.syntax().text_range(), buf);
+            let replacement_start = module_ast
+                .syntax()
+                .children_with_tokens()
+                .find(|child| {
+                    !matches!(
+                        child.kind(),
+                        syntax::SyntaxKind::COMMENT
+                            | syntax::SyntaxKind::ATTR
+                            | syntax::SyntaxKind::WHITESPACE
+                    )
+                })
+                .unwrap_or_else(|| syntax::SyntaxElement::Node(module_ast.syntax().clone()))
+                .text_range()
+                .start();
```

Comment:
It would probably be simpler to just take the start of the `mod` token's syntax range

Reply:
Yeah, you are right. I thought it didn't work, because of visibility modifiers, but those are just a special case that can be removed when using the mod token, actually simplifying the code